plugins/opds.koplugin/opdsbrowser.lua

```diff
@@ -804,12 +856,16 @@ function OPDSBrowser:showDownloads(item)
             end
         end
     end
-    if (#type_buttons % 2 == 1) then -- we need even number of type buttons
-        table.insert(type_buttons, {text = ""})
-    end
-    for i = 2, #type_buttons, 2 do
-        table.insert(buttons, {type_buttons[i - 1], type_buttons[i]}) -- type buttons, two in a row
+    -- handles adding download button(s)
+    if (download_buttons ~= {}) then
+        if (#download_buttons == 1) then
+            table.insert(buttons, download_buttons)
+            table.insert(buttons, {})  -- add separator for stream buttons
+        elseif (#download_buttons % 2 == 1) then
+            table.insert(download_buttons, {text = ""})
+        end
     end
+    table.insert(buttons, stream_buttons)
```

Comment:
It will produce additional separator when there are no stream buttons.
You can declare `local stream_buttons` and then initialize it as an array of two buttons if the stream exists.

Reply:
What If i just implement a check to see if any stream buttons exist?
```
    if (stream_buttons ~= {}) then
        table.insert(buttons, stream_buttons)
        table.insert(buttons, {}) -- separator
    end
```

Reply:
> * download buttons, but no stream buttons

There will be no separator for 2 download buttons case.

Reply:
I Gotcha, now the separator is after the logic that determines if there are more than 1 download button. Now, if there is any number of download buttons there will be a separator.

Reply:
It seems you've missed to insert download buttons (more than 1 case).

Reply:
The _One_ Time I feel confident in a solution!  :)

I'll fix it.

Reply:
yeah, so I read `table.insert(download_buttons, {text = ""})` as `table.insert(buttons, download_buttons)`, which is why I forgot to add `table.insert(buttons, download_buttons)` in the more than 1 case.